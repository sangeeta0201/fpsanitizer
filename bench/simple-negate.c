#include <stdio.h>
#include <math.h>

int main() {
  double x = 1e-20;
  double y = (- x);
  double z = 5.0;
  double w = (y + z) - z;
  printf("%e\n", w);
}
