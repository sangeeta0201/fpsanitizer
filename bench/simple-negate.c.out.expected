(output
  (argIdx 0)
  (function "main")
  (filename "simple-negate.c")
  (line-num 9)
  (instr-addr 40054C)
  (avg-error 61.901585)
  (max-error 61.901585)
  (num-calls 1)
  (influences
    (
    (
     (expr
       (FPCore ()
          (- (+ 5.000000 (- 10.000000e-21)) 5.000000)))
     (var-problematic-ranges)
     (example problematic input ())
     (function "main")
     (filename "simple-negate.c")
     (line-num 8)
     (instr-addr 400530)
     (avg-error 61.901585)
     (max-error 61.901585)
     (avg-local-error 61.901585)
     (max-local-error 61.901585)
     (num-calls 1))
    )
  )
)
